geoh5py/objects/surveys/electromagnetics/base.py

```diff
@@ -545,6 +567,38 @@ def _edit_validate_property_groups(
             if value.name not in self.metadata["EM Dataset"]["Property groups"]:
                 self.metadata["EM Dataset"]["Property groups"].append(value.name)
 
+    def _fetch_transmitter_id(self) -> ReferencedData | IntegerData | None:
+        """
+        Utility method to retrieve the transmitter ID property, either from
+        metadata or from list of children Data.
+        """
+        if self.type == "Receivers":
+            uid = self.metadata["EM Dataset"].get("Tx ID property", None)
+        else:
+            uid = self.metadata["EM Dataset"].get("Tx ID tx property", None)
+
+        if uid is not None:
+            tx_id_property = self.get_entity(uid)[0]
+        else:
+            tx_id_property = self.get_entity("Transmitter ID")[0]
+            # Reset the metadata
+
+        if not isinstance(tx_id_property, ReferencedData | IntegerData | type(None)):
```

Comment:
Seems like `type(none)` shouldn't be in the list.  You're asking for `getattr` on `tx_id_property` later

Reply:
It has to be there since the get_entity can return None, which means that the tx ID is not defined on the object (possible since not mandatory)